Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -2557,77 +2578,134 @@ private void HandleTickSize(object sender, IB.TickSizeEventArgs e)
         {
             Symbol symbol;
 
-            if (!_subscribedTickets.TryGetValue(e.TickerId, out symbol)) return;
+            if (!_subscribedTickets.TryGetValue(e.TickerId, out symbol))
+            {
+                return;
+            }
 
-            var tick = new Tick();
-            // in the event of a symbol change this will break since we'll be assigning the
-            // new symbol to the permtick which won't be known by the algorithm
-            tick.Symbol = symbol;
             var securityType = symbol.ID.SecurityType;
-            tick.Quantity = AdjustQuantity(securityType, e.Size);
-            tick.Time = GetRealTimeTickTime(symbol);
-
-            if (tick.Quantity == 0) return;
+            var quantity = AdjustQuantity(securityType, e.Size);
 
+            Tick tick;
             switch (e.Field)
             {
                 case IBApi.TickType.BID_SIZE:
                 case IBApi.TickType.DELAYED_BID_SIZE:
 
-                    tick.TickType = TickType.Quote;
+                    if (!_lastQuoteTicks.TryGetValue(symbol, out tick))
+                    {
+                        // tick size message must be preceded by a tick price message
+                        return;
+                    }
 
-                    _lastBidPrices.TryGetValue(symbol, out tick.BidPrice);
-                    _lastBidSizes[symbol] = (int)tick.Quantity;
+                    tick.BidSize = quantity;
 
-                    tick.Value = tick.BidPrice;
-                    tick.BidSize = tick.Quantity;
+                    if (tick.BidPrice == 0)
+                    {
+                        // no bid price, do not emit tick
+                        return;
+                    }
+
+                    if (tick.BidPrice > 0 && tick.AskPrice > 0 && tick.BidPrice >= tick.AskPrice)
+                    {
+                        // new bid price jumped above previous ask price, wait for new ask price
```

Comment:
Another idea 🤔  would it be worth emitting half a bid quote here, setting the ask values to 0?

Reply:
Hmm... point 2 of this PR was to actually prevent us from emitting incomplete/half ticks when possible 😃 

In these normal bid/ask price changes, i.e. when both bid and ask quotes move up or down **and** are both non-zero, we will always receive both half-quotes in sequence.